morph_tool/transform.py

```diff
@@ -98,3 +104,128 @@ def align(section, direction):
+            if method == 'trunk':
+
+                if target_point is not None:
+                    target_secid = point_to_section_segment(morph, target_point)[0] - 1
+                    if target_secid is None:
+                        return None
+                elif neurite_type == 'apical':
+                    target_secid = apical_point_section_segment(morph)[0]
+                else:
+                    raise Exception(f"We don't know how to get target point for {neurite_type}.")
+
+                return np.vstack(
+                    [section.points
+                     for section in morph.sections[target_secid].iter(IterType.upstream)]
+                )
+            if method == 'first_section':
+                return root_section.points
+
+            if method == 'first_segment':
+                return root_section.points[:2]
+
+            return np.vstack([section.points for section in root_section.iter()])
+
+
+def _get_principal_direction(points):
+    '''Return the principal direction of a point cloud
+    It is the eigen vector of the covariance matrix with the highest eigen value.
+
+    Taken frorm neuror.unravel.'''
+    X = np.copy(np.asarray(points))
+    X -= np.mean(X, axis=0)
+    C = np.dot(X.T, X)
+    w, v = np.linalg.eig(C)
+    return v[:, w.argmax()]
+
+
+def align_morphology(
+    morph, direction=None, method='whole', neurite_type='apical', target_point=None
+):
+    """In-place alignment of a morphology towards a 'direction'.
+
+    The base algorithm is based on eigenvalue decomposition of the correlation matrix obtained
+    from points in specified neurites, giving the principal axis of the neurite.
+    Currently, five algorithms are implemented, differing in the choice of points:
+
+    1) with method='whole': All the points in the apical dendrite are used.
+    2) with method='trunk': Points in section up to the tarrget points are used.
+    3) with method='first_section': Only the points in the first section are used.
+    4) with method='first_segment': Only the points in the first segment are used.
+    5) with method an ndarray or list, we will use it as the direction directly
+
+    If no neurite is present, no rotation is applied, and the identity rotation is returned.
+    If two neurites of same types are present, the first accessed by Morphio will be used.
+
+    Args:
+        morph (morphio.Morphology): morphology to align
+        direction (ndarray): 3-vector for final direction, if None, [0, 1, 0] will be used
+        method (str): method for alignment.
+        neurite_type (str): neurite to consider, can only be apical or axon
+        target_point (ndarray): position of target point for method='trunk',
+            if None and neurite_type='apical', it will be estimated
+
+    Returns:
+        3x3 array with applied rotation matrix
+    """
+    if isinstance(method, str) and method not in [
+        'whole', 'trunk', 'first_section', 'first_segment'
```

Comment:
lets move those constants into variables/enum, so they can be reused here and in `_get_points`. Also I don't see that 'whole' is used in `_get_points`

Reply:
I'm not familiar with this enum thing, could you give me a quick example? thanks!

Reply:
```python
from enum import Enum


class AlignMethod(Enum):
    WHOLE = 'whole'
    TRUNK = 'trunk'
    FIRST_SECTION = 'first_section'
    FIRST_SEGMENT = 'first_segment'

    @classmethod
    def values(cls):
        return list(map(lambda c: c.value, cls))

#...

    if method == AlignMethod.TRUNK.value:

#...

    if isinstance(method, str) and method not in AlignMethod.values():
        raise NotImplementedError(f"Method {method} is not implementd")

```

Reply:
very fancy!